### PR TITLE
Project 4: Hooks

### PR DESCRIPTION
**Pull request by crgc**
### Features

- Refactored all class-based components into functional components

### CODE REVIEW

File: src/components/Calculator.js [hunk 1]
```diff
@@ -1,28 +1,20 @@
...
-}
+  return (
+    <div className="Calculator">
+      <Display value={displayOutput} />
+      <Keyboard onKeyPress={updateDisplay} />
+    </div>
+  );
+};
 
 export default Calculator;
```
Comment (DeliceLydia) [inline src/components/Calculator.js:20]: - [x] Please work on the `%` functionality currently its not working

**Review by DeliceLydia — CHANGES_REQUESTED** [review]
# Hi @crgc , 

Good job so far! There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._


### Cheers and Happy coding!👏👏👏

Feel free to leave a question, or a comment on the PR thread if something is not clear 100%. 

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/components/Calculator.js (hunk 1 above)
Comment (crgc) [reply src/components/Calculator.js:20]: What exactly is "not working"? - I just tested it and it works fine. Please be more specific or provide a screenshot.

**Comment by crgc** [comment]
Next reviewer please be advised: the mod operation `%` is working perfectly. I don't know what the previous reviewer is referring to, since she did not provide any details. Saying something is "not working" when it is obviously working is not a proper review.

File: src/components/Calculator.js (hunk 1 above)
Comment (DeliceLydia) [reply src/components/Calculator.js:20]: ### Hello @crgc,

We had a confusion about if we can check `%` as a percentage or `modulus
before I was checking percentage but got clarifications that it is modulus for this
project! I apologize for the inconvenience.

Lydia

**Review by DeliceLydia — APPROVED** [review]
# Congratulations🎉 🎉 

### Hi @crgc , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



### Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard).. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**